### PR TITLE
Reorganize ExtractMetadata

### PR DESCRIPTION
**Pull request by turian**
This just moves everything around in a more logical order.

Everything you might override is at the top. This is used by #24 to create a smaller diff

### CODE REVIEW

**Review by jorshi — APPROVED** [review]

**Comment by jorshi** [comment]
Do you know why we're getting a key error on slug?

File: heareval/tasks/pipeline.py [hunk 1]
```diff
@@ -300,58 +340,20 @@ def split_train_test_val(self, metadata: pd.DataFrame):
...
+        if self.task_config["embedding_type"] == "event":
+            assert "start" in df.columns
+            assert "end" in df.columns
+        return df
+
     def run(self):
         # Process metadata gets all metadata to be used for the task
         metadata = self.get_all_metadata()
 
-        # Deterministically shuffle the metadata
```
Comment (jorshi) [inline heareval/tasks/pipeline.py:307]: Looks like this all got moved out into a separate function `postprocess_all_metadata`, but that function is never called

**Review by jorshi — COMMENTED** [review]
Approved a bit too early -- see comment in code below

File: heareval/tasks/pipeline.py (hunk 1 above)
Comment (turian) [reply heareval/tasks/pipeline.py:307]: oups caught that in the upcoming PR #23

**Comment by turian** [comment]
https://github.com/neuralaudio/hear-preprocess/pull/30

**Pull request closed without merging**